src/briefcase/integrations/subprocess.py

```diff
@@ -413,14 +416,18 @@ def _stream_output_thread(self, popen_process):
 
         :param popen_process: popen process to stream stdout
         """
-        while True:
-            # readline should always return at least a newline (ie \n)
-            # UNLESS the underlying process is exiting/gone; then "" is returned
-            output_line = ensure_str(popen_process.stdout.readline())
-            if output_line:
-                self.command.logger.info(output_line)
-            elif output_line == "":
-                return
+        # ValueError is raised if stdout is unexpectedly closed.
+        # This can happen if the user starts spamming CTRL+C, for instance.
+        # Silently exit to avoid Python printing the exception to the console.
+        with suppress(ValueError):
```

Comment:
I found this while experimenting with sending CTRL+C while building an AppImage in Docker. Different versions of Python manifest the error differently in the console but they all raise `ValueError`.

Reply:
Can we add a test case for this as well?

Reply:
Let me know if you think this test goes too far...or isn't testing the right thing. I didn't just want to mock `readline()` to raise `ValueError` and instead verify that using `readline()` on closed IO actually led to a `ValueError`.

Both `io.StringIO` and `io.TextIOWrapper` (which is used for `stdout`) both inherit from `io.TextIOBase`....which is where I'm assuming this `ValueError` comes from....if not deeper.

Reply:
Definitely agreed that it's better to try and cause a ValueError rather than just raise one as a mock; the problem is that in my testing, this tests still passes when the `suppress(ValueError)` is disabled - I suspect this is because some interaction of pytest and threading causes the `ValueError` to be hidden from the console. However, after poking around, I can't come up with a better test for this, so I'm happy to accept it as is.

Reply:
Oh yeah; that's a good point....the error only surfaces in the console when Python is exiting.

I came up with this change to the test....its a little janky but seems to work:

```python
def test_stdout_closes_unexpectedly(mock_sub, popen_process, monkeypatch, capsys):
    """Streamer silently exits from ValueError because stdout was closed."""

    class suppress:
        is_valueerror_exception_raised = []

        def __init__(self, *exceptions):
            self._exceptions = exceptions

        def __enter__(self):
            pass

        def __exit__(self, exctype, excinst, exctb):
            self.is_valueerror_exception_raised.append(exctype is ValueError)
            return exctype is not None and issubclass(exctype, self._exceptions)

    def monkeypatch_ensure_str(value):
        """Close stdout when ensure_str() runs on output from readline()."""
        popen_process.stdout.close()
        return value

    popen_process.stdout = StringIO(initial_value="output line 1\noutput line 2")
    monkeypatch.setattr(subprocess, "ensure_str", monkeypatch_ensure_str)
    monkeypatch.setattr(subprocess, "suppress", suppress)

    mock_sub.stream_output("testing", popen_process)

    assert capsys.readouterr().out == "output line 1\n"
    assert suppress.is_valueerror_exception_raised == [True]
```

I haven't figured out why `is_valueerror_exception_raised` needs to be a mutable type...but that was the only thing I could quickly get working.

Reply:
Yeah - that's... janky :-) I'm not sure it's worth it as a test improvement; it's monkey patching the object that we're trying to verify is working.